ui/pages/confirmation/confirmation.js

```diff
@@ -398,6 +416,11 @@ export default function ConfirmationPage({
           )
           ///: END:ONLY_INCLUDE_IN
         }
+        {prevSnapInfo && pendingConfirmation?.origin === 'metamask' && (
+          ///: BEGIN:ONLY_INCLUDE_IN(snaps)
+          <SnapAuthorshipHeader snapId={prevSnapInfo.origin} />
```

Comment:
Rather than adding custom logic here in the generic component, the `success` and `error` methods support an optional `header` property which lets you define any additional components you want to put in the header. This should work fine with the `SnapAuthorishipHeader` also.

```
{
	  message: `**${snapName}** is now available to use.`,
	  header: [{ key: 'snap', element: 'SnapAuthorship', props: { snapId } }],
}
```

Reply:
I tried to get this working but with no success. I messaged you some of my issues and also outlined the code I wrote along with the issues in this [draft PR](https://github.com/MetaMask/metamask-extension/pull/20995). For now I will move on.

Reply:
New PR to migrate to this better pattern: https://github.com/MetaMask/metamask-extension/pull/21076/files